src/protocol/prf_sharding/bucket.rs

```diff
@@ -0,0 +1,234 @@
+                        ctx.set_total_records(1),
+                        RecordId::from(0),
+                        breakdown_key_share,
+                        value_share,
+                        count,
+                        robust,
+                    )
+                    .await
+                    .unwrap()
+                },
+            )
+            .await
+            .reconstruct()
+    }
+
+    #[test]
+    fn semi_honest_move_in_range() {
+        run(|| async move {
+            let mut rng = thread_rng();
+            let count = rng.gen_range(1..MAX_BREAKDOWN_COUNT);
+            let breakdown_key = rng.gen_range(0..count);
+            let mut expected = vec![Fp32BitPrime::ZERO; count];
+            expected[breakdown_key] = Fp32BitPrime::truncate_from(VALUE);
+
+            let result = move_to_bucket(count, breakdown_key, false).await;
+            assert_eq!(result, expected, "expected value at index {breakdown_key}");
+        });
+    }
+
+    #[test]
+    fn semi_honest_move_in_range_robust() {
+        run(|| async move {
+            let mut rng = thread_rng();
+            let count = rng.gen_range(1..MAX_BREAKDOWN_COUNT);
+            let breakdown_key = rng.gen_range(0..count);
+            let mut expected = vec![Fp32BitPrime::ZERO; count];
+            expected[breakdown_key] = Fp32BitPrime::truncate_from(VALUE);
+
+            let result = move_to_bucket(count, breakdown_key, true).await;
+            assert_eq!(result, expected, "expected value at index {breakdown_key}");
+        });
+    }
+
+    #[test]
+    fn semi_honest_move_out_of_range() {
+        run(move || async move {
+            let mut rng: rand::rngs::ThreadRng = thread_rng();
+            let count = rng.gen_range(2..MAX_BREAKDOWN_COUNT - 1);
+            let breakdown_key = rng.gen_range(count..MAX_BREAKDOWN_COUNT);
+
+            let result = move_to_bucket(count, breakdown_key, false).await;
+            assert_eq!(result.len(), count);
+            assert_eq!(
+                result.into_iter().sum::<Fp32BitPrime>(),
+                Fp32BitPrime::truncate_from(VALUE)
+            );
+        });
+    }
+
+    #[test]
+    fn semi_honest_move_out_of_range_robust() {
+        run(move || async move {
+            let mut rng: rand::rngs::ThreadRng = thread_rng();
+            let count = rng.gen_range(2..MAX_BREAKDOWN_COUNT - 1);
+            let breakdown_key = rng.gen_range(count..MAX_BREAKDOWN_COUNT);
+
+            let result = move_to_bucket(count, breakdown_key, true).await;
+            assert_eq!(result.len(), count);
+            assert!(result.into_iter().all(|x| x == Fp32BitPrime::ZERO));
+        });
+    }
+
+    #[test]
+    #[should_panic]
+    fn move_out_of_range_too_many_buckets_type() {
+        run(move || async move {
+            _ = move_to_bucket(MAX_BREAKDOWN_COUNT + 1, 0, false).await;
+        });
+    }
+
+    #[test]
+    #[should_panic]
+    fn move_out_of_range_too_many_buckets_steps() {
```

Comment:
Arent both these functions checking max breakdown count > MAX_NUMBER?

Reply:
The first should fail on the first check (`count > (1<< BK::BITS)`).

The second should fail on the second check (`count > 128`).  We can increase that once #799 lands)

Reply:
oh ok. Sounds good